title_belt_nhl/schedule.py

```diff
@@ -139,23 +140,32 @@ def find_match(self, current_belt_holder, from_date) -> Match:
                 match.belt_holder = current_belt_holder
                 return match
 
-    def find_nearest_path_str(self, teams, path_string, from_date=None) -> str:
-        newTeams = []
-        if from_date:
-            self.set_from_date(from_date)
-        for tm in teams:
-            splits = tm.split(" -> ")
-            cur_match: Match = self.find_match(splits[-1], self.from_date)
+    def find_nearest_path_v2(self, scenarios: list[list[Match]]) -> list[Match]:
+        newScenarios: list[list[Match]] = []
+
+        for s in scenarios:
+            cur_match = s[-1]
             if cur_match:
                 if cur_match.away == self.team or cur_match.home == self.team:
-                    return f"{tm} -> {cur_match}"
-                newTeams.append(f"{tm} -> {cur_match} -> {cur_match.away}")
-                newTeams.append(f"{tm} -> {cur_match} -> {cur_match.home}")
-
-        path_string = self.find_nearest_path_str(
-            newTeams, path_string, cur_match.serial_date
-        )
-        return path_string
+                    # found a path to team
+                    return s
+
+                # Add new Scenario branches
+                newScenarios.append(
+                    self.createNewScenarioBranch(cur_match.home, cur_match.serial_date, s)
+                )
+                newScenarios.append(
+                    self.createNewScenarioBranch(cur_match.away, cur_match.serial_date, s)
+                )
+
+        shortestPath = self.find_nearest_path_v2(newScenarios)
+        return shortestPath
```

Comment:
What happens if the season ends and no match is found?

Reply:
We get paid.

Reply:
Attempted to handle this [here](https://github.com/kawa2287/title-belt-nhl/blob/d4fc48bc18d55d9daa3f7bba275dd2b9f367891f/title_belt_nhl/schedule.py#L154C1-L160C24).   And added tests [here](https://github.com/kawa2287/title-belt-nhl/blob/d4fc48bc18d55d9daa3f7bba275dd2b9f367891f/title_belt_nhl/tests/test_schedule.py#L173C1-L199C36)